lib/instrument/headers.js

```diff
@@ -0,0 +1,260 @@
+          cursor += 4;
+          const archived_data = headerBuffer.slice(cursor, cursor + strLen);
+          data.push(unarchive(archived_data));
+          cursor += strLen;
+          break;
+        }
+        case 3:
+          data.push(headerBuffer.readUInt32LE(cursor));
+          cursor += 4;
+          break;
+        case 4:
+          data.push(headerBuffer.readBigUInt64LE(cursor));
+          cursor += 8;
+          break;
+        case 6:
+          data.push(headerBuffer.readBigInt64LE(cursor));
+          cursor += 8;
+          break;
+      }
+    }
+    return data;
+  }
+}
+
+
+function InstrumentRPCParseError (data) {
+  this.data = data ;
+}
+
+
+class DTXMessageAuxBuffer {
+  constructor () {
+    this._buf = Buffer.alloc(0);
+  }
+  appendInt (value) {
+    const buf = Buffer.alloc(12);
+    buf.writeUInt32LE(10, 0);
+    buf.writeUInt32LE(3, 4);
+    buf.writeUInt32LE(value, 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+
+  }
+  appendLong (value) {
+    const buf = Buffer.alloc(16);
+    buf.writeUInt32LE(10, 0);
+    buf.writeUInt32LE(4, 4);
+    // eslint-disable-next-line no-undef
+    buf.writeBigUInt64LE(BigInt(value), 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+
+  }
+  appendSignedInt (value) {
+    const buf = Buffer.alloc(12);
+    buf.writeUInt32LE(10, 0);
+    buf.writeUInt32LE(3, 4);
+    buf.writeInt32LE(value, 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+  }
+  appendSignedLong (value) {
+    const buf = Buffer.alloc(16);
+    this._buf.writeUInt32LE(10, 0);
+    this._buf.writeUInt32LE(6, 4);
+    // eslint-disable-next-line no-undef
+    this._buf.writeBigInt64LE(BigInt(value), 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+
+  }
+  appendObject (value) {
+    const buf = archive(value);
+    const buf2 = Buffer.alloc(12);
+    buf2.writeUInt32LE(10, 0);
+    buf2.writeUInt32LE(2, 4);
+    buf2.writeUInt32LE(buf.length, 8);
+    this._buf = Buffer.concat([this._buf, buf2, buf]);
+
+  }
+  getBytes () {
+    const out = Buffer.alloc(16);
+    out.writeUInt32LE(0x01F0, 0);
+    out.writeUInt32LE(this._buf.length, 8);
+    return Buffer.concat([out, this._buf]);
+  }
+}
+
+
+class DTXMessage {
+  constructor () {
+    this._buf = undefined;
+    this._messageHeader = undefined;
+    this._payloadHeader = undefined;
+    this._identifier = undefined;
+    this._channelCode = undefined;
+    this._expectsReply = undefined;
+    this._selector = undefined;
+    this._conversationIndex = 0;
+    this._flags = FLAGS_TYPE;
+    this.auxiliaries = new DTXMessageAux();
+  }
+  get identifier () {
+    return this._messageHeader.identifier;
+  }
+  get channelCode () {
+    return this._messageHeader.channel;
+  }
+  get expectsReply () {
+    return this._messageHeader.expectsReply;
+  }
+  get selector () {
+    return this._selector;
+  }
+  get conversationIndex () {
+    return this._messageHeader.conversationIndex;
```

Comment:
such getter might throw exceptions if the corresponding properties are undefined. It would be safer to use the following format: `return this._messageHeader?.conversationIndex;`